tests/contrib/compare_ids_fail_test.bzl

```diff
@@ -16,6 +16,17 @@
 
 def _impl(ctx):
     test_code = """ '
+py_binary(
```

Comment:
why are you need this duplicated from the BUILD file above here?
If its because your test is rewriting the file contrib/BUILD, then do not re write it, write a new BUILD file in some subdirectory (e.g., contrib/fail_test/BUILD)

Reply:
Changed it to copy the BUILD instead of duplicate code